tests/CMakeLists.txt

```diff
@@ -15,6 +15,11 @@ if(NOT PERL_FOUND)
     message(FATAL_ERROR "Cannot build test suites without Perl")
 endif()
 
+# Enable definition of various functions used throughout the testsuite
+# (hostname, strdup, fileno...) even when compiling with -std=c99. Harmless
+# on non-POSIX platforms.
+add_definitions("-D_POSIX_C_SOURCE=200809L")
```

Comment:
This must also be added in the Makefile.

This could be provided in `tests/scripts/generate_code.pl` for the consistency - having all the feature macros in the source files. However I do not have a strong opinion on this.